packages/components/src/unit-control/test/index.tsx

```diff
@@ -87,35 +104,42 @@ describe( 'UnitControl', () => {
 			const input = getInput();
 			const select = getSelect();
 
-			expect( input ).toBeTruthy();
-			expect( select ).toBeTruthy();
+			expect( input ).toBeInTheDocument();
+			expect( select ).toBeInTheDocument();
 		} );
 
 		it( 'should render custom className', () => {
-			render( <UnitControl className="hello" /> );
+			const { container: withoutClassName } = render( <UnitControl /> );
 
-			const el = getComponent();
+			const { container: withClassName } = render(
+				<UnitControl className="hello" />
+			);
 
-			expect( el.classList.contains( 'hello' ) ).toBe( true );
+			expect( withoutClassName.firstChild ).toMatchDiffSnapshot(
+				withClassName.firstChild
+			);
 		} );
 
 		it( 'should not render select, if units are disabled', () => {
 			render( <UnitControl value="3em" units={ [] } /> );
 			const input = getInput();
-			const select = getSelect();
+			// Using `queryByRole` instead of `getSelect` because we need to test
+			// for this element NOT to be in the document.
+			const select = screen.queryByRole( 'combobox' );
 
-			expect( input ).toBeTruthy();
-			expect( select ).toBeFalsy();
+			expect( input ).toBeInTheDocument();
+			expect( select ).not.toBeInTheDocument();
 		} );
 
 		it( 'should render label if single units', () => {
 			render( <UnitControl units={ [ { value: '%', label: '%' } ] } /> );
 
-			const select = getSelect();
-			const label = getUnitLabel();
+			const select = screen.queryByRole( 'combobox' );
+			// The unit is not being displayed!
+			const label = screen.getByText( '%' );
```

Comment:
Currently this line is failing in unit tests — that's because we switched from just checking that an element with class `components-unit-control__unit-label` exists, to actually looking for the unit label being displayed (`testing-library` FTW!)

I'm not sure if we should call this a bug, though. Technically we're not passing any `value` to `UnitControl`, and therefore the component internally has not `value` to parse a unit from.

If our expectations are that `UnitControl` would display the `%` unit, for how the component is coded right now, it would mean that effectively the selected unit is `%` (which would also be reflected in the new components's `value` and probably fired through the `change` callbacks). Would that be a correct behavior to expect? I'm personally not so sure.

Reply:
Huh, good investigation! I would expect that when there is only one unit, the unit would be visible at mount and stay visible no matter what. This kind of seems like a common use case so I'd say it's worth fixing if it isn't too hard. What do you think?